add-on/src/lib/ipfs-client/index.js

```diff
@@ -55,46 +56,32 @@ async function destroyIpfsClient (browser) {
   if (!client) return
   try {
     await client.destroy(browser)
-    await _reloadIpfsClientDependents(browser) // sync (API stopped working)
+    await reloadIpfsClientDependents(browser) // sync (API stopped working)
   } finally {
     client = null
   }
 }
 
-function _isWebuiTab (url) {
-  const bundled = !url.startsWith('http') && url.includes('/webui/index.html#/')
-  const ipns = url.includes('/webui.ipfs.io/#/')
-  return bundled || ipns
-}
-
-function _isInternalTab (url, extensionOrigin) {
-  return url.startsWith(extensionOrigin)
-}
-
-async function _reloadIpfsClientDependents (browser, instance, opts) {
+async function reloadIpfsClientDependents (
+  browser, instance, opts, reloadExtensions = [WebUiReloader, LocalGatewayReloader, InternalTabReloader]) {
   // online || offline
   if (browser.tabs && browser.tabs.query) {
     const tabs = await browser.tabs.query({})
     if (tabs) {
-      const extensionOrigin = browser.runtime.getURL('/')
-      tabs.forEach((tab) => {
-        // detect bundled webui in any of open tabs
-        if (_isWebuiTab(tab.url)) {
-          log(`reloading webui at ${tab.url}`)
-          browser.tabs.reload(tab.id)
-        } else if (_isInternalTab(tab.url, extensionOrigin)) {
-          log(`reloading internal extension page at ${tab.url}`)
-          browser.tabs.reload(tab.id)
-        }
-      })
+      const reloadChecks = await prepareReloadExtensions(reloadExtensions, browser, log)
+      reloadChecks.forEach(check => check.reload(tabs))
     }
   }
+
   // online only
   if (client && instance && opts) {
     // add important data to local ipfs repo for instant load
     setTimeout(() => precache(instance, opts), 5000)
   }
 }
 
-exports.initIpfsClient = initIpfsClient
-exports.destroyIpfsClient = destroyIpfsClient
+module.exports = {
+  initIpfsClient,
+  destroyIpfsClient,
+  reloadIpfsClientOfflinePages: (...args) => reloadIpfsClientDependents(...args, [LocalGatewayReloader])
```

Comment:
nit: I would prefer this be defined above

I would also prefer explicit args (`(a1, a2, a3) => reloadIpfsClientDependents(a1, a2, a3, [LocalGatewayReloader])`) since we don't have typescript to help with argument safety.

Reply:
Done!